converters/pokemon.py

```diff
@@ -288,20 +298,56 @@ def convert_pdata(input_csv, header=DEFAULT_HEADER):
         evolve = Evolve(header, pokemon_list)
         filter_data = FilterData(header)
         index_order = IndexOrder(header)
+        
+        poke_by_name = {}
 
+        # Collect all the rows into Pokemon types
         for index, row in enumerate(reader, 1):
             if not row:
                 continue
 
             # Each row is one Pokemon
             poke = Pokemon(header)
             poke.setup(row)
+            poke_by_name[poke.name] = poke
+
+        # Some rows are variants of a single pokemon type. Let's go collect those
+        variant_by_type = {}
+        def_poke_by_type = {}
+        def_type_by_variant = {}
```

Comment:
Would be nice to have the names a bit clearer.
`def` is a keyword in Python which makes the name confusing. A "Bulbasaur" isn't a type of a Pokemon it is a species.

```python
variant_by_species = {}
default_poke_by_species = {}
default_species_by_variant = {}
```